lib/build.js

```diff
@@ -979,6 +1010,27 @@ const buildActs = (config, act_proofs) => {
 
 }
 
+// apply every where binding in turn to every other where binding
+const expandWhereBindings = (bindings) => {
+  for (i = 0; i < bindings.length; i++) {
+    bindings = bindings.map(b => ({
+      from: b.from,
+      to: b.to.replace(bindings[i].from, bindings[i].to)
+    }))
+  }
+  return bindings
+}
+
+// apply where bindings to the spec
+const applyWhereBindings = (spec, bindings) => {
+  if (typeof bindings === "undefined") return spec
```

Comment:
Nit: I think it would be nice to move this check to an `extractWhereBindings` function, so that the "meaty" functions could only focus on the actual logic.

Reply:
done.